src/java/org/apache/cassandra/tools/nodetool/JmxConnect.java

```diff
@@ -0,0 +1,239 @@
+     * @return The exit code.
+     */
+    public static int executionStrategy(ParseResult parseResult)
+    {
+        CommandSpec jmx = parseResult.commandSpec().mixins().get(MIXIN_KEY);
+        if (jmx == null)
+            throw new InitializationException("No JmxConnect command found in the top-level hierarchy");
+
+        try (JmxConnectionCommandInvoker invoker = new JmxConnectionCommandInvoker((JmxConnect) jmx.userObject()))
+        {
+            return invoker.execute(parseResult);
+        }
+        catch (JmxConnectionCommandInvoker.CloseException e)
+        {
+            jmx.commandLine()
+               .getErr()
+               .println("Failed to connect to JMX: " + e.getMessage());
+            return jmx.commandLine().getExitCodeExceptionMapper().getExitCode(e);
+        }
+    }
+
+    /**
+     * Initialize the JMX connection to the Cassandra node using the provided options.
+     */
+    @Override
+    protected void execute(NodeProbe probe)
+    {
+        assert probe == null;
+        try
+        {
+            if (isNotEmpty(username))
+            {
+                if (isNotEmpty(passwordFilePath))
+                    password = readUserPasswordFromFile(username, passwordFilePath);
+
+                if (isEmpty(password))
+                    password = promptAndReadPassword();
+            }
+
+            probe(username.isEmpty() ? nodeProbeFactory.create(host, parseInt(port))
+                                     : nodeProbeFactory.create(host, parseInt(port), username, password));
+        }
+        catch (IOException | SecurityException e)
+        {
+            Throwable rootCause = Throwables.getRootCause(e);
+            output.printError("nodetool: Failed to connect to '%s:%s' - %s: '%s'.%n", host, port,
+                         rootCause.getClass().getSimpleName(), rootCause.getMessage());
+            throw new InitializationException("Failed to connect to JMX", e);
+        }
+    }
+
+    @Override
+    public void close() throws Exception
+    {
+        if (probe() == null)
+            return;
+        ((AutoCloseable) probe()).close();
+    }
+
+    private static String readUserPasswordFromFile(String username, String passwordFilePath)
+    {
+        String password = EMPTY;
+
+        File passwordFile = new File(passwordFilePath);
+        try (Scanner scanner = new Scanner(passwordFile.toJavaIOFile()).useDelimiter("\\s+"))
+        {
+            while (scanner.hasNextLine())
+            {
+                if (scanner.hasNext())
+                {
+                    String jmxRole = scanner.next();
+                    if (jmxRole.equals(username) && scanner.hasNext())
+                    {
+                        password = scanner.next();
+                        break;
+                    }
+                }
+                scanner.nextLine();
+            }
+        }
+        catch (FileNotFoundException e)
+        {
+            throw new RuntimeException(e);
```

Comment:
nit: Might be a good place for `UncheckedIOException`

Reply:
Fixed.